### PR TITLE
Cli params

### PR DESCRIPTION
**Pull request by lbalaceanu**
Testing the code review feature via pull request.

### CODE REVIEW

File: daemon/main.c [hunk 1]
```diff
@@ -82,6 +82,23 @@ static char *spooldir;
...
+static char *graphitep = NULL;
+static char *graphite_prefix_s = NULL;
+static char *redisps = NULL;
+static char *redisps_write = NULL;
+static char *log_facility_cdr_s = NULL;
+static char *log_facility_rtcp_s = NULL;
+static int sip_source = 0;
+static char *homerp = NULL;
+static char *homerproto = NULL;
+struct rtpengine_config_params rtp_initial_cfg;
```
Comment (lbalaceanu) [inline daemon/main.c]: Global variables are avoided. Try to make your own variable static.

File: daemon/main.c [hunk 2]
```diff
@@ -625,7 +654,50 @@ static void create_everything(struct main_context *ctx) {
 	timeval_from_us(&tmp_tv, (long long) graphite_interval*1000000);
 	set_graphite_interval_tv(&tmp_tv);
 }
-
+void fill_rtpengine_cfg(struct rtpengine_config_params* cfg_param) {
+	struct rtpengine_config_params config_params;
```
Comment (lbalaceanu) [inline daemon/main.c]: This variable is redundant, you could directly allocate values to cfg_params.

File: daemon/call.h [hunk 3]
```diff
@@ -438,6 +472,52 @@ struct callmaster {
 
 	struct callmaster_config conf;
 	struct timeval          latest_graphite_interval_start;
+
+};
+struct rtpengine_config_params {
```
Comment (lbalaceanu) [inline daemon/call.h]: If you like you could give a name to this type: struct rtpengine_config_params {...} rtpengine_config_params_t so you don't use the `struct rtpengine_config_params` every time.

File: daemon/main.c [hunk 4]
```diff
@@ -625,7 +654,50 @@ static void create_everything(struct main_context *ctx) {
...
+	config_params.table = table;
+	config_params.no_fallback = no_fallback;
+	config_params.interface = if_a;
```
Comment (lbalaceanu) [inline daemon/main.c]: You should do deep copying when pointers are involved.

File: daemon/call.h (hunk 3 above)
Comment (hdikme) [reply daemon/call.h]: Done!

File: daemon/main.c [hunk 5]
```diff
@@ -654,49 +671,104 @@ static void create_everything(struct main_context *ctx) {
...
+	GList* l;
+	struct intf_config* gptr_data;
+	gpointer gptr;
+	char **itr;
+	char **itr2;
+	int len1 = 0, len2 = 0;
+	for (itr = if_a; *itr; itr++) {
+		++len1;
+	}
+	itr2 = cfg_param->cfg_addition.interface_name = (char**)malloc(sizeof(char*)*len1 + 1);
```
Comment (lbalaceanu) [inline daemon/main.c]: (sizeof(char*)*len1 + 1) -- why the +1 ?

File: daemon/main.c [hunk 6]
```diff
@@ -654,49 +671,104 @@ static void create_everything(struct main_context *ctx) {
...
+	for (itr = if_a; *itr ; itr++) {
+		*itr2 = strdup(*itr);
+		itr2++;
+		if(!*itr2) {
```
Comment (lbalaceanu) [inline daemon/main.c]: What is the purpose of this check?

File: daemon/main.c [hunk 7]
```diff
@@ -654,49 +671,104 @@ static void create_everything(struct main_context *ctx) {
...
+	itr2 = cfg_param->cfg_addition.subscribe_ks = (char**)malloc(sizeof(char*)*len2 + 1);
+	for (itr = ks_a ; *itr ; itr++) {
+		*itr2 = strdup(*itr);
+		itr2++;
+		if(!*itr2) {
+			break;
+		}
+	}
+
+	for(l=interfaces.head; l ; l=l->next) {
```
Comment (lbalaceanu) [inline daemon/main.c]: Whenever you consider shallow copy, try this: g_queue_copy ().

File: daemon/cli.c [hunk 8]
```diff
@@ -698,104 +707,102 @@ static void cli_incoming_params_current(char* buffer, int len, struct callmaster
...
-		printlen = snprintf(replybuffer,(outbufend-replybuffer), "redis_expires\n, %d, -> %d",rtp_initial_cfg.redis_expires, m->conf.redis_expires_secs);
+	if(m->p_initial_config->redis_expires != m->conf.redis_expires_secs) {
+		printlen = snprintf(replybuffer,(outbufend-replybuffer), "redis_expires\n, %d, -> %d",m->p_initial_config->redis_expires, m->conf.redis_expires_secs);
 		ADJUSTLEN(printlen,outbufend,replybuffer);
 	}
 
-
-	c = m->conf.cfg_addition.interfaces.head;
+	c = m->conf.cfg_addition.interface_ips.head;
 	ifa2 = c->data;
```
Comment (lbalaceanu) [inline daemon/cli.c]: What not make for that iterates through both m->conf.cfg_addition.interface_ips and m->p_initial_config->cfg_addition.interface_ips at the same type?

File: daemon/cli.c [hunk 9]
```diff
@@ -609,6 +609,302 @@ static void cli_incoming_list(char* buffer, int len, struct callmaster* m, char*
...
+			"rec_method = %s \nrec_format = %s \n",
+			rtp_initial_cfg.listen_tcp, rtp_initial_cfg.listen_udp, rtp_initial_cfg.listen_ng, rtp_initial_cfg.listen_cli,
+			rtp_initial_cfg.graphite, rtp_initial_cfg.graphite_prefix, rtp_initial_cfg.redis, rtp_initial_cfg.redis_write,
+			rtp_initial_cfg.b2b_url, rtp_initial_cfg.log_facility_cdr, rtp_initial_cfg.log_facility_rtcp, rtp_initial_cfg.homer,
+			rtp_initial_cfg.homer_protocol,  rtp_initial_cfg.rec_dir, rtp_initial_cfg.rec_method, rtp_initial_cfg.rec_format);
+	ADJUSTLEN(printlen,outbufend,replybuffer);
+	return ;
+}
+
+static void cli_incoming_params_current(char* buffer, int len, struct callmaster* m, char* replybuffer, const char* outbufend) {
```
[the remark this answers is not in the document]
Comment (hdikme) [reply daemon/cli.c]: As you said they do pretty similar stuff but they'r using different structs and hence there are so many different parameters.  And dont know if it ll make much difference, but if u have any suggestions i d be glad to know :)

File: daemon/main.c (hunk 5 above)
Comment (hdikme) [reply daemon/main.c]: for the NULL pointer.

File: daemon/main.c (hunk 6 above)
Comment (hdikme) [reply daemon/main.c]: The for loop has been used to deep copy interface_names. but the if check seems unnecessary, will be removed, thx :+1:

File: daemon/main.c (hunk 7 above)
Comment (hdikme) [reply daemon/main.c]: sure but in this case i needed a deep copy.

File: daemon/call.h [hunk 10]
```diff
@@ -398,25 +398,27 @@ struct callmaster_cfg_addition {
...
+	char log_facility_rtcp[MAX_SIZE];
+	char log_facility_cdr[MAX_SIZE];
+	char homer[MAX_SIZE];
+	char homer_protocol[MAX_SIZE];
+	char rec_dir[MAX_SIZE];
+	char rec_method[MAX_SIZE];
+	char rec_format[MAX_SIZE];
+
+	GQueue interface_ips ;
+	GQueue keyspaces;
```
[the remark this answers is not in the document]
Comment (hdikme) [reply daemon/call.h]: Changed it as u suggested, and the "keyspaces" has been replaced by "keyspaces_default"

File: daemon/cli.c [hunk 11]
```diff
@@ -609,51 +609,60 @@ static void cli_incoming_list(char* buffer, int len, struct callmaster* m, char*
        ADJUSTLEN(printlen,outbufend,replybuffer);
    }
 }
-static void cli_incoming_params_start(char* replybuffer, const char* outbufend) {
-	int printlen, count = 0;
-	GList *l;
+
+static void cli_incoming_params_start(char* buffer, int len, struct callmaster* m, char* replybuffer, const char* outbufend) {
```
[the remark this answers is not in the document]
Comment (hdikme) [reply daemon/cli.c]: Done.

**Pull request closed without merging**